www/js/rover-api.js

```diff
@@ -0,0 +1,98 @@
+/*----- MISC GLOBALS -----*/
+var roverDomain = ''; //Later, if the rover is not hosting the webpage, its address will go here
+var roverApiPath = '/api/v1/';
+
+/*----- HELPER FUNCTIONS -----*/
+function roverResource(resource) {
+	return roverDomain+roverApiPath+resource;
+}
+
+/*----- ROVER API FUNCTIONS -----*/
+function sendMotorCommand(command, pin, speed) {
+	$.post(roverResource('sendcommand'),
+		{
+			command: command,
+			pin: pin,
+			speed: Number(speed)
+		}, function (response) {
+			//writeToConsole(response);
+		});
+}
+
+function saveDesign() {
+	xml = Blockly.Xml.workspaceToDom(workspace);
+	xmlString = Blockly.Xml.domToText(xml);
+	$.post(roverResource('blockdiagrams'), {bdString: xmlString, designName: designName}, function(response){
+	}).error(function(){
+			writeToConsole("There was an error saving your design to the rover");
+	});
+}
+
+function refreshSavedBds() {
+	$.get(roverResource('blockdiagrams'), function(json){
+		if (!json.result.length){
+			$('#savedDesignsArea').text("There are no designs saved on this rover");
+		} else {
+			$('#savedDesignsArea').empty();
+			json.result.forEach(function(entry) {
+				$('#savedDesignsArea').append("<a href='#' class='button' style='margin:10px;' onclick='return loadDesign(\""+entry+"\")'>"+entry+"</a>");
+			});
+		}
+	}
+	);
+}
+
+function loadDesign(name) {
+	$('#loadModal').foundation('reveal', 'close');
+	$.get(roverResource('blockdiagrams')+'/'+name, function(response){
+		workspace.clear();
+
+		xmlDom = Blockly.Xml.textToDom(response.getElementsByTagName('bd')[0].childNodes[0].nodeValue);
+		Blockly.Xml.domToWorkspace(workspace, xmlDom);
+		if (name == 'event_handler_hidden')
+			designName = "Unnamed_Design_" + (Math.floor(Math.random()*1000)).toString();
+		else
+			designName = name;
+		$('a#downloadLink').attr("href", "saved-bds/"+designName+".xml");
+		$('a#downloadLink').attr("download", designName+".xml");
+		$('a#designNameArea').text(designName);
+
+		hideBlockByComment("MAIN EVENT HANDLER LOOP");
+		var hiddenBlock;
+		var allBlocksHidden = true;
+		for (hiddenBlock of blocksToHide) {

```

Comment:
It's an array in mission-control.js. I moved these functions over from mission-control.js, but there are still lots of entanglements. At some point it will be good to go detangle them.